src/components/asset-list/RecyclerAssetList2/FastComponents/FastBalanceCoinRow.tsx

```diff
@@ -0,0 +1,330 @@
+              <View style={[cx.row, cx.bottom]}>
+                <View style={cx.textWrapper}>
+                  <Text
+                    align="left"
+                    color={{ custom: theme.colors.blueGreyDark50 }}
+                    numberOfLines={1}
+                    size="14px"
+                  >
+                    {nativeDisplay ?? ''}
+                  </Text>
+                </View>
+
+                <Text color={{ custom: changeColor }} size="14px">
+                  {percentageChangeDisplay}
+                </Text>
+              </View>
+            </View>
+          </View>
+        </ButtonPressAnimation>
+      </View>
+    );
+  }
+);
+
+MemoizedBalanceCoinRow.displayName = 'MemoizedBalanceCoinRow';
+
+export default React.memo(function BalanceCoinRow({
+  uniqueId,
+  extendedState,
+}: {
+  uniqueId: string;
+  extendedState: ExtendedState;
+}) {
+  const {
+    theme,
+    nativeCurrencySymbol,
+    navigate,
+    nativeCurrency,
+    hiddenCoins,
+    pinnedCoins,
+    toggleSelectedCoin,
+    isCoinListEdited,
+  } = extendedState;
+
+  const onPress = useCallback(() => {
+    toggleSelectedCoin(uniqueId);
+  }, [uniqueId, toggleSelectedCoin]);
+
+  // HACK to make sure we don't rerender MemoizedBalanceCoinRow
+  // when isCoinListEdited === true and we need to change onPress callback
+  const maybeCallback = useRef<null | (() => void)>(null);
+  maybeCallback.current = isCoinListEdited ? onPress : null;
+
+  // we return false to make sure we don't run includes for each row when it's not needed
+  const isHidden = isCoinListEdited ? hiddenCoins[uniqueId] : false;
+  const isPinned = isCoinListEdited ? pinnedCoins[uniqueId] : false;
+
+  return (
+    <View style={[cx.rootContainer, !isCoinListEdited && cx.nonEditMode]}>
+      {isCoinListEdited && (
+        <CoinCheckButton
+          isHidden={isHidden}
+          isPinned={isPinned}
+          onPress={onPress}
+          theme={theme}
+          uniqueId={uniqueId}
+        />
+      )}
+
+      <MemoizedBalanceCoinRow
+        isHidden={isHidden}
+        maybeCallback={maybeCallback}
+        nativeCurrency={nativeCurrency}
+        nativeCurrencySymbol={nativeCurrencySymbol}
+        navigate={navigate}
+        theme={theme}
+        uniqueId={uniqueId}
+      />
+    </View>
+  );
+});
+
+const cx = StyleSheet.create({
+  bottom: {
+    marginTop: 12,
+  },
+  checkboxContainer: {
+    width: 51,
+  },
+  checkboxInnerContainer: {
+    alignContent: 'center',
+    alignItems: 'center',
+    alignSelf: 'center',
+    height: 40,
+    justifyContent: 'center',
+    width: 51,
+  },
+  checkmarkBackground: {
+    ...borders.buildCircleAsObject(22),
+    ...padding.object(4.5),
+    backgroundColor: colors.appleBlue,
+    left: 19,
+    position: 'absolute',
+  },
+  circleOutline: {
+    ...borders.buildCircleAsObject(22),
+    borderWidth: 1.5,
+    left: 19,
+    position: 'absolute',
+  },
+  coinIconFallback: {
+    backgroundColor: '#25292E',
```

Comment:
Should this color depend on the theme? if yes then we have constant `greyDark, dark and shadow` with the same value

Reply:
Was not used